sceptre/template_handlers/file.py

```diff
@@ -0,0 +1,168 @@
+
+from sceptre.exceptions import UnsupportedTemplateFileTypeError, TemplateSceptreHandlerError
+from sceptre.template_handlers import TemplateHandler
+
+
+class File(TemplateHandler):
+    """
+    Template handler that can load files from disk. Supports JSON, YAML, Jinja2 and Python.
+    """
+
+    def __init__(self, *args, **kwargs):
+        self.logger = logging.getLogger(__name__)
+        super(File, self).__init__(*args, **kwargs)
+
+    def schema(self):
+        return {
+            "type": "object",
+            "properties": {
+                "path": {"type": "string"},
+            },
+            "required": ["path"]
+        }
+
+    def handle(self):
+        file_extension = os.path.splitext(self.arguments["path"])[1]
+        path = self.arguments["path"]
+        try:
+            if file_extension in {".json", ".yaml", ".template"}:
+                with open(path) as template_file:
+                    return template_file.read()
+            elif file_extension == ".j2":
+                return self._render_jinja_template(
+                    os.path.dirname(path),
+                    os.path.basename(path),
+                    {"sceptre_user_data": self.sceptre_user_data}
+                )
+            elif file_extension == ".py":
+                return self._call_sceptre_handler()
+            else:
+                raise UnsupportedTemplateFileTypeError(
+                    "Template has file extension %s. Only .py, .yaml, "
+                    ".template, .json and .j2 are supported.",
+                    os.path.splitext(path)[1]
+                )
+        except Exception as e:
+            self._print_template_traceback()
+            raise e
+
+    def _call_sceptre_handler(self):
+        """
+        Calls the function `sceptre_handler` within templates that are python
+        scripts.
+
+        :returns: The string returned from sceptre_handler in the template.
+        :rtype: str
+        :raises: IOError
+        :raises: TemplateSceptreHandlerError
+        """
+        # Get relative path as list between current working directory and where
+        # the template is
+        # NB: this is a horrible hack...
+        path = self.arguments["path"]
+
+        relpath = os.path.relpath(path, os.getcwd()).split(os.path.sep)
+        relpaths_to_add = [
+            os.path.sep.join(relpath[:i + 1])
+            for i in range(len(relpath[:-1]))
+        ]
+        # Add any directory between the current working directory and where
+        # the template is to the python path
+        for directory in relpaths_to_add:
+            sys.path.append(os.path.join(os.getcwd(), directory))
+        self.logger.debug(
+            "%s - Getting CloudFormation from %s", self.name, path
+        )
+
+        if not os.path.isfile(path):
+            raise IOError("No such file or directory: '%s'", path)
+
+        import imp
+        module = imp.load_source(path, path)
```

Comment:
@zaro0508 As we discussed in the slack, the integration test issue is related to stack caching when the stack has the same name.

Summary:
The scenario `Generate template using a valid python template file` runs before the `Generating erroneous python templates` and both uses the same stack name in the scenario, and when the `Generate template using a valid python template file` load the valid template python module, the `imp.load_source` method was using the `imp.load_source(self.name, path)` which caches the python module per stack name, causing this issue.

To solve that I just change the `imp.load_source` to the path as a name as well. so the module will cache based on the path, which is unique.

I tested locally and works fine to me.

Reply:
very nice sleuthing @lucasvieirasilva.  very much appreciated.  while looking at that i noticed that `imp` library has been deprecated so i did an update in PR #1092